plugins/publish-changes/admin/src/index.js

```diff
@@ -0,0 +1,52 @@
+import pluginPkg from "../../package.json";
+import pluginId from "./pluginId";
+import App from "./containers/App";
+import Initializer from "./containers/Initializer";
+import lifecycles from "./lifecycles";
+
+export default (strapi) => {
+  const pluginDescription =
+    pluginPkg.strapi.description || pluginPkg.description;
+  const icon = pluginPkg.strapi.icon;
+  const name = pluginPkg.strapi.name;
+
+  const plugin = {
+    blockerComponent: null,
+    blockerComponentProps: {},
+    description: pluginDescription,
+    icon,
+    id: pluginId,
+    initializer: Initializer,
+    injectedComponents: [],
+    isReady: false,
+    isRequired: pluginPkg.strapi.required || false,
+    layout: null,
+    lifecycles,
+    mainComponent: App,
+    name,
+    preventComponentRendering: false,
+    trads: {},
+    menu: {
+      pluginsSectionLinks: [
+        {
+          destination: `/plugins/${pluginId}`,
+          icon,
+          label: {
+            id: `${pluginId}.plugin.name`,
+            defaultMessage: name,
+          },
+          name,
+          permissions: [
+            // Uncomment to set the permissions of the plugin here
+            // {
+            //   action: '', // the action name should be plugins::plugin-name.actionType
+            //   subject: null,
+            // },
+          ],
```

Comment:
We should make sure that our action is limited to admins. I would assume that requires something here.

Reply:
Not quite sure how to resolve this. I was able to restrict the backend routes, but it logs the user out when the user has insufficient permissions, which isn't optimal. In the client I would be able to restrict the actions if I could edit them for this plugin.

As you can see here this plugin is not visible in the list of plugins.
<img width="1014" alt="Screenshot 2022-06-01 at 0 31 46" src="https://user-images.githubusercontent.com/56825922/171293876-84cc223e-355c-4266-8916-017b077b4601.png">